docs/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix.md

```diff
@@ -0,0 +1,615 @@
+
+Next, you should update the string resources:
+
+* Change `AnalyzerTitle` to "Variable can be made constant".
+* Change `AnalyzerMessageFormat` to "Can be made constant".
+* Change `Description` to "Make Constant".
+
+You can see the update resources in the following figure:
+
+![Update string resources](media/how-to-write-csharp-analyzer-code-fix/update-string-resources.png)
+
+Also, change the `Category` to "Usage" as shown in the following code:
+```C#
+private const string Category = "Usage";
+```
+
+When you're finished, the code in DiagnosticAnalyzer.cs should look like the following code.
+
+```C#
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace FirstAnalyzerCS
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class MakeConstAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "MakeConstCS";
+        private const string Title = "Variable can be made constant";
+        private const string MessageFormat = "Can be made constant";
+        private const string Description = "Make Constant";
+        private const string Category = "Usage";
+
+        private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.LocalDeclarationStatement);
+        }
+
+        private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            throw new NotImplementedException();
+        }
+    }
+}
+```
+
+### Spot declarations that are already const
+
+Now you're ready to write the logic to determine whether a local variable can be declared as a const in the AnalyzeNode method. You'll make these changes in steps as you learn about the APIs available to write code that understands code.
+
+Start by performing some syntactic analyzer when your action is invoked. Open **MakeConst.cs**. In the `AnalyzeNode` method, cast the node passed in to the <xref:Microsoft.CodeAnalysis.CSharp.Syntax.LocalDeclarationStatementSyntax> type. You can safely assume this cast will succeed because the registered action specifies that type of node.
+
+```C#
+var localDeclaration = (LocalDeclarationStatementSyntax)context.Node;
```

Comment:
The `generate missing method` actually named my parameter as `obj` not `context`.